### ISSUE TITLE
Model overfitting for 20 classes ( PASCAL VOC 2007 + 2012 dataset )

### ISSUE THREAD

**Issue by 100daggers** [aladdinpersson/Machine-Learning-Collection#25]
Hi Aladdin , Thank you so much for your video and explanations,  
I am currently doing a project on object detection , and your video helped me a lot.
thank you once again.

I  have a problem of overfitting in the model . I am getting test map as 10% , train map 90% . I trained on PASCAL VOC 2007 + VOC 2012 data.
I have tried every way I could think of to reduce the overfitting ( dropout layer , weight decay , added 5k more images ,data augmentation , used pretrained extraction weights ,step LR etc etc ) , tried everything as close as possible to original paper 
its been a month now and I am still not  able to figure out why . could you please help me?  ( I have used your code for everything). it would be a great help if you can suggest something with respect to your code .

**P.S : I used the same code and modified it for 2 classes and 5 classes , I have got good results , 2classes : test map 50% , 5 classes : test map 60%.**

**Comment by aladdinpersson** [comment]
I'm not sure what the problem could be, that sounds very strange. What kind of data augmentation are you using? I would try using quite extensive augmentation and probably use albumentations library. When I find some time I can take a look at this in more depth and see what performance I am able to get.

**Comment by jbalzer12** [comment]
I also got the problem that with this implementation I reach a pretty good mAP on the training data (VOC 2007 and 2012) after 135 epochs. But when I evaluate the model training on the test data I get a very (!) bad mAP... I read this multiple times (e.g. here: https://www.kaggle.com/code/vexxingbanana/yolov1-from-scratch-pytorch/comments) while I made some research to solve this topic. Unfortunately I wasn't able to do so yet... Maybe some of you guys got an idea?
